respokeSDK/src/main/java/com/digium/respokesdk/RespokeClient.java

```diff
@@ -511,6 +531,170 @@ public RespokeGroup getGroup(String groupIDToFind) {
+     * @param completionListener The callback called when this async operation has completed
+     */
+    public void getGroupHistories(final ArrayList<String> groupIds, final Integer maxMessages,
+                                  final GroupHistoriesCompletionListener completionListener) {
+        if (!isConnected()) {
+            getGroupHistoriesError(completionListener, "Can't complete request when not connected" +
+                    "Please reconnect!");
+            return;
+        }
+
+        if ((maxMessages == null) || (maxMessages < 1)) {
+            getGroupHistoriesError(completionListener, "maxMessages must be at least 1");
+            return;
+        }
+
+        if ((groupIds == null) || (groupIds.size() == 0)) {
+            getGroupHistoriesError(completionListener, "At least 1 group must be specified");
+            return;
+        }
+
+        Uri.Builder builder = new Uri.Builder();
+        builder.appendQueryParameter("limit", maxMessages.toString());
+        for (int i = 0; i < groupIds.size(); i++) {
+            builder.appendQueryParameter("groupIds", groupIds.get(i));
+        }
+
+        String urlEndpoint = "/v1/group-histories" + builder.build().toString();
+        signalingChannel.sendRESTMessage("get", urlEndpoint, null,
+                new RespokeSignalingChannel.RESTListener() {
+            @Override
+            public void onSuccess(Object response) {
+                if (!(response instanceof JSONObject)) {
+                    getGroupHistoriesError(completionListener, "Invalid response from server");
+                    return;
+                }
+
+                final JSONObject json = (JSONObject) response;
+                final HashMap<String, ArrayList<RespokeGroupMessage>> results = new HashMap<>();
+
+                for (Iterator<String> keys = json.keys(); keys.hasNext();) {
+                    final String key = keys.next();
+
+                    try {
+                        final JSONArray jsonMessages = json.getJSONArray(key);
+
+                        final ArrayList<RespokeGroupMessage> messageList = new ArrayList<>();
+
+                        for (int i = 0; i < jsonMessages.length(); i++) {
+                            final JSONObject jsonMessage = jsonMessages.getJSONObject(i);
+                            final RespokeGroupMessage message = buildGroupMessage(jsonMessage);
+                            messageList.add(message);
+                        }
+
+                        results.put(key, messageList);
+                    } catch (JSONException e) {
+                        getGroupHistoriesError(completionListener, "Error parsing JSON response");
+                        return;
+                    }
+                }
+
+                new Handler(Looper.getMainLooper()).post(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (null != completionListener) {
```

Comment:
This C'ism looks a little odd here. Normally I'd expect the ```null``` literal to be on the right-hand side of any expression where the language strongly discourages/forbids assignment inside a conditional expression. Any reason you aren't writing this as:

```
if (completionListener != null) {
    ...
}
```